src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs

```diff
@@ -449,6 +449,13 @@ private StrongNameKeys ComputeStrongNameKeys()
             // when both attributes and command-line options specified, cmd line wins.
             string keyFile = _compilation.Options.CryptoKeyFile;
 
+            // Public sign requires a keyfile
+            if (DeclaringCompilation.Options.PublicSign)
+            {
+                // If we're public signing, we don't need a strong name provider
+                return StrongNameKeys.Create(keyFile, MessageProvider.Instance);
```

Comment:
I don't think it is the right thing to do to silently ignore the AssemblyKeyFileAttributefor for public sign. Either we should support the attribute and report an error for a relative path in case of public signing, or we should report an error that the attribute cannot be used in combination with public sign.